### PR TITLE
[WIP] git-node: prompt and conditionally auto push

### PR DESCRIPTION
**Pull request by refack**
Seeking feedback and testing guidance.

Refs: https://github.com/nodejs/node/pull/23941#issuecomment-435678992

### CODE REVIEW

**Comment by refack** [comment]
I just saw `cli.prompt`, but anyway I needed a more complicated user input.

File: lib/landing_session.js [hunk 1]
```diff
@@ -227,10 +243,20 @@ class LandingSession extends Session {
...
-    cli.log('To finish landing:');
-    cli.log(`1. Run \`git push ${upstream} ${branch}\``);
-    const url = `https://github.com/${owner}/${repo}/pull/${prid}`;
-    cli.log(`2. Post in ${url}: \`Landed in ${willBeLanded}\``);
+    const pushCommandArgs = ['push', upstream, branch];
+    const prUrl = `https://github.com/${owner}/${repo}/pull/${prid}`;
+    const landComment = `Post in ${prUrl}: \`Landed in ${willBeLanded}\``;
+    const autoLand = await asyncQuestion(prid);
+    if (autoLand) {
+      await runAsync('git', pushCommandArgs);
```
Comment (joyeecheung) [inline lib/landing_session.js]: If I remember correctly...a failed `git push` child process should exit with 1 and make this process also exit with 1 (through runAsync)?

Comment (refack) [reply lib/landing_session.js]: AFAICT `runAsync(x, y, { ignoreFailure:false })` will exit the process.

Comment (joyeecheung) [inline lib/landing_session.js]:
```suggestion
      await runAsync('git', pushCommandArgs, {ignoreFailure: false});
```

**Comment by targos** [comment]
@refack `inquirer` is now part of the dependencies (in case that helps for this PR)

**Comment by joyeecheung** [comment]
@targos Maybe open a separate issue about refactoring cli.prompt on top of inquirer ?

**Comment by targos** [comment]
@JoyeeCheung sure. I thought this PR was blocked on prompt-related things. My bad.
Is it ready for review then?

**Comment by joyeecheung** [comment]
@targos I thought this was blocked on the TODO in the comment?

**Comment by codebytere** [comment]
Can this be closed?

**Comment by mmarchini** [comment]
Closing, if anyone wants to reopen feel free to do so.

**Pull request closed without merging**